File: src/main/java/com/github/redis/RedisClusterConfigProperties.java
package com.github.redis;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.util.List;

/**
 * Created by zk_chs on 16/4/11.
 */
@Component
@ConfigurationProperties(prefix = "spring.redis.cluster")
public class RedisClusterConfigProperties {

    /*
     * spring.redis.cluster.nodes[0] = 127.0.0.1:7379
     * spring.redis.cluster.nodes[1] = 127.0.0.1:7380
     * ...
     */
    public List<String> nodes;

    public List<String> getNodes() {
        return nodes;
    }

    public void setNodes(List<String> nodes) {
        this.nodes = nodes;
    }
}

File: src/main/scala/com/github/config/RedisClusterConfigProperties.scala
package com.github.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

import scala.language.postfixOps

/**
  * Created by zk_chs on 16/4/14.
  */
@Component
@ConfigurationProperties(prefix = "spring.redis.cluster")
class RedisClusterConfigProperties {

  var nodes: java.util.List[java.lang.String] = _

  def getNodes = nodes

  def setNodes(nodes: java.util.List[java.lang.String]) {this nodes = nodes}

}
